Moodle401 fixes

sarahjcotton opened the pull request:
Fixes #197 and #199

sarahjcotton committed:
Fix output debug info error

sarahjcotton committed:
Fix for issue #197

brendanheywood commented on renderer.php:
```diff
@@ -251,7 +251,7 @@ function envbar_close(el) {
      * @return string Debug text
      */
-    protected function get_debug_text($canedit = false, $config) {
+    protected function get_debug_text($config, $canedit = false) {
```
was there a reason for swapping the order of these args?

sarahjcotton replied:
Apparently optional params should come last in php 8+
https://www.php.net/manual/en/migration80.deprecated.php

Fixes:
Deprecated: Required parameter $config follows optional parameter $canedit in
local/envbar/renderer.php on line 254

Or would you prefer to drop the default value instead?

brendanheywood replied:
Nope that makes perfect sense thanks

brendanheywood commented:
@sarahjcotton this looks ok code wise but a few bits of house keeping:

- ci is still red for 3.10
- can you bump the version and supports metadata
- and it is using the old shared workflows. Are you able to convert that over as part of getting this green?

https://github.com/catalyst/catalyst-moodle-workflows#add-the-workflow

sarahjcotton committed:
Version bump

dmitriim commented on version.php:
```diff
@@ -29,9 +29,9 @@
 $plugin->component = 'local_envbar';
 $plugin->maturity  = MATURITY_STABLE;
-$plugin->supported = [310, 311];      // A range of branch numbers of supported moodle versions.
+$plugin->supported = [401, 401];      // A range of branch numbers of supported moodle versions.
```
I can't see any specific changes for 4.1 exclusively. Why are we dropping support for 310 and 311 here? And also bumping a min version in line 34?

sarahjcotton committed:
CI fixes

sarahjcotton commented:
I'm hoping this is ready to go now. I've updated the badge in the README and changed the ci workflow to hopefully the right one as well as fixed the ci issues.
